### PR TITLE
Fixed #27888 -- Added link to clear all filters in the admin changelist view.

### PR DESCRIPTION
**Pull request by hramezani**
[Ticket 27888](https://code.djangoproject.com/ticket/27888)
Complete the [PR](https://github.com/django/django/pull/8156), Added tests and release note.

### CODE REVIEW

**Comment by felixxm** [comment]
@hramezani Thanks :+1: However, tt still doesn't look good on different resolutions, e.g.
obraz
and completely disappears on mobiles, e.g.
obraz

I would prefer to add it to the `changelist-filter` on the right side.

**Comment by hramezani** [comment]
@felixxm If we want to add it to the `changelist-filter`, we will miss the button when we have `search_fields` and don't have `list_filter`.

**Comment by hramezani** [comment]
I changed it. and put the button on the `actions` section. Here are the button in different cases:

- Both `search_fields` and `list_filter` are enable:



- `search_fields`  is enable:



- `list_filter` is enable:



- And here is the mobile view:

**Comment by felixxm** [comment]
@hramezani Putting "Clear filters" in actions is misleading, IMO.
> @felixxm If we want to add it to the changelist-filter, we will miss the button when we have search_fields and don't have list_filter.

I think it's fine, because search fields has an option to clear filters (see [comment](https://code.djangoproject.com/ticket/27888#comment:1)). I would add _"Clear all filters"_ in a sidebar, e.g.
obraz

File: django/contrib/admin/templates/admin/change_list.html [hunk 1]
```diff
@@ -61,6 +61,13 @@
           <div id="changelist-filter">
             <h2>{% translate 'Filter' %}</h2>
             {% for spec in cl.filter_specs %}{% admin_list_filter cl spec %}{% endfor %}
+            {% if cl.has_filters or cl.search_fields%}
+              {% if cl.preserved_filters %}
+              <h3>
+                <a href="?{% if cl.is_popup %}_popup=1{% endif %}">&#10006; {% trans 'Clear all filters' %}</a>
```
Comment (felixxm) [inline django/contrib/admin/templates/admin/change_list.html]: `trans` -> `translate`

**Comment by felixxm** [comment]
@hramezani Thanks :+1: I moved link to the top of sidebar (this should increase visibility) and added small style changes:

obraz

@carltongibson :point_up: What do you think?

**Comment by carltongibson** [comment]
Screenshot looks good to me. (Didn't look at the rest of the PR.)

**Comment by frnhr** [comment]
Hi all! Please check out this PR https://github.com/django/django/pull/12924 (and ticket https://code.djangoproject.com/ticket/31597), I think I found a bug with the new link. Fixed it as well.

**Pull request merged**